### ISSUE TITLE
update to ffmpeg 5.1

### ISSUE THREAD

**Issue by derhuerst** [eugeneware/ffmpeg-static#110]
[ffmpeg 5.1 has been released.](https://git.ffmpeg.org/gitweb/ffmpeg.git/tag/4f3c925bd5e8f253c1dbf57eb9552b92ce0b12d5)

@BtbN @sudo-nautilus FYI our nightly build script doesn't pull the latest 5.0.1 release yet.

- [ ] Linux x64
- [ ] Linux x86
- [ ] Linux `arm`
- [ ] Linux `arm64`
- [x] macOS x64
- [ ] macOS arm64
- [x] Windows x64
- [x] Windows x86
- [ ] FreeBSD (cc @TheFrank)

**Comment by Thefrank** [comment]
Good news: https://github.com/Thefrank/ffmpeg-static-freebsd/releases/tag/v5.1 and it passes FATE

Not  good news: many of the libraries used in the `--enable-libxxxxxxx` are older versions due either to lack of patches in ports or build failures in 5.1

Neutral news: Still targets the EOL FreeBSD 12.2 but works under FreeBSD 13.0+

**Comment by whyboris** [comment]
I've been looking forward to the `JPEG XL` aka `.jxl` format to be supported by _FFmpeg_ and it [now is](https://www.phoronix.com/news/FFmpeg-JPEG-XL-Lands) 🥳 starting with version `5.1` 🎉 

Would be _wonderful_ to have this repository upgrade 🙇 at your convenience ❤️ thank you!

**Comment by derhuerst** [comment]
> Would be _wonderful_ to have this repository upgrade 🙇 at your convenience ❤️ thank you!

I'm waiting for people to build & upload binaries.

**Comment by whyboris** [comment]
I have an `m1` Mac - is building a binary something you think is within my capabilities? If so, where could I upload the binary for this project to be able to use it?

**Comment by derhuerst** [comment]
> I have an `m1` Mac - is building a binary something you think is within my capabilities? If so, where could I upload the binary for this project to be able to use it?

I don't know much about compiling ffmpeg, so I can't really answer that question, sorry. Does someone else know?

The [project where `ffmpeg-static` pulls the M1/arm64 binary from](https://www.osxexperts.net) has a shell script, but I'm not sure if some of the hacks are still necessary.

**Comment by piranna** [comment]
There are newer versions just of this weekend, what we would need to move this forward?

Also, does it makes sense to have a nightly script to review sources and publish new versions automatically? I have already done some of them in the past.

**Comment by derhuerst** [comment]
> what we would need to move this forward?

Last time I checked, osxexperts.net didn't have ffmpeg 5.1 & 5.1.1 binaries yet, now it does (but not 5.1.2 yet). This means that I will publish two versions of `ffmpeg-static`, using ffmpeg 5.1 & 5.1.1, respectively.

> Also, does it makes sense to have a nightly script to review sources and publish new versions automatically? I have already done some of them in the past.

I think some automation to get notified about the latest available binaries would be helpful. Publishing by hand is not that much work and much less error-prone.

**Comment by derhuerst** [comment]
> Last time I checked, osxexperts.net didn't have ffmpeg 5.1 & 5.1.1 binaries yet, now it does (but not 5.1.2 yet). This means that I will publish two versions of `ffmpeg-static`, using ffmpeg 5.1 & 5.1.1, respectively.

Turns out that the Linux 5.1 binaries are not available *anymore* (under URLs I know of). I have contacted the person hosting them. This does not apply to the Linux 5.1.1 binaries.

**Comment by piranna** [comment]
> Turns out that the Linux 5.1 binaries are not available _anymore_ (under URLs I know of). I have contacted the person hosting them. This does not apply to the Linux 5.1.1 binaries.

Then, maybe go directly for 5.1.1 if it's available for all platforms?

**Comment by derhuerst** [comment]
I have updated my initial post, to my knowledge it lists those binaries still missing.

cc @sudo-nautilus @Thefrank

**Comment by Thefrank** [comment]
I have 6.0 on my repo now.

**Comment by GenoD** [comment]
Hi @derhuerst it seems 6.0 has been released for all binaries. Any chance for a release in the near future? Thanks for your work here!

**Comment by derhuerst** [comment]
@sudo-nautilus I couldn't find an ffmpeg 6.0 release binary for Windows x86. Is there one?

**Comment by derhuerst** [comment]
I have updated the list to include `ffprobe` as well, in order to support #19 via #113.

Currently, we're waiting for either Linux ([johnvansickle.com](https://johnvansickle.com/ffmpeg/)), macOS arm64 ([osxexperts.net](https://www.osxexperts.net)), Windows x86 ([repo](https://github.com/sudo-nautilus/FFmpeg-Builds-Win32/releases) by @sudo-nautilus) or FreeBSD ([repo](https://github.com/Thefrank/ffmpeg-static-freebsd/releases) by @Thefrank), depending on the ffmpeg release.

**Comment by Thefrank** [comment]
https://github.com/Thefrank/ffmpeg-static-freebsd/releases/tag/v5.1.2

built but forgot to publish it. currently have it as "pre-release" as I lack the original build environment to test it

**Comment by sbsrnt** [comment]
@derhuerst is this the one https://github.com/sudo-nautilus/FFmpeg-Builds-Win32/releases/tag/latest 👀?

**Comment by derhuerst** [comment]
> @derhuerst is this the one https://github.com/sudo-nautilus/FFmpeg-Builds-Win32/releases/tag/latest 👀?

Yes, except that it uses the tag `latest`, which will break with the next build. The download script needs a (as) permanent (as possible) link.

**Comment by Makio64** [comment]
If ffmpeg 6.0 is ready, what about making a branch for it ? I would like to use this version especially for some new encodings

**Comment by paulrouget** [comment]
Not sure if this is useful, but I thought I'd mention that I have a build script for ffmpeg & ffprobe that properly builds on Windows (i686 & x86_64), Linux (i686 & x86_64) and Mac (Intel & Arm). Builds are automatically tested. The builds are "as static as possible" (on system dependencies allowed) (list of dynamic dependencies: https://github.com/paulrouget/ffmpeg-static-builder/blob/94f88ad51f10525397be7604bbb451e3d8310eb2/recipes/ffmpeg.sh#L68-L103).

I only build what I personally need, but I'm willing to add more options if that helps. All dependencies are direct submodules of the repo.

I will be adding recent build here: https://github.com/paulrouget/ffmpeg-static-builder/releases

**Comment by derhuerst** [comment]
I have published `ffmpeg-static@5.2.0`, which downloads ffmpeg 6.0 binaries. Thanks everyone! 🎉
